src/shared/sam/cli/samCliLocalInvoke.ts

```diff
@@ -113,6 +115,22 @@ export class DefaultSamLocalInvokeCommand implements SamLocalInvokeCommand {
             }
         })
 
+        if (params.timeout) {
+            await Promise.race([debuggerPromise, params.timeout.timer]).then( async (value) => {
+                if (value === false) {
```

Comment:
This feels too arbitrary, and doesn't bode well for doing something similar in scenarios where timeout is used with other promises that also return boolean. Is there another way to determine which promise won the race?

Reply:
I think what we want to know is "did `debuggerPromise` complete/resolve?" There should be a clean way of achieving this.

Reply:
I don't think there's an easy, built-in way to determine if a promise has completed, hence the boolean value. The other alternative is to have the timer promise reject on completion and handle it the same way as if the other promise rejected as well (that process would also need to be killed--we don't do this currently).

Reply:
This is super crude, but what about something along the lines of:

let x: boolean = false // "did debuggerPromise complete?"
debuggerPromise = debuggerPromise.then(() => x = true).catch(() => x = true)

...

Promise.race(...).then(() => check x)

Reply:
Expanded the scope of the `promiseResolved` boolean to cover this case for now. I'll look into extending Promise as well; if that works, this might not be necessary.

Reply:
✔️